### PR TITLE
Update cranelift-08-09.md

### PR DESCRIPTION
**Pull request by avanhatt**
Lots of technical stuff outside my wheelhouse this week, so feel free to fix up!

### CODE REVIEW

File: cranelift/2023/cranelift-08-09.md [hunk 1]
```diff
@@ -11,12 +11,58 @@
...
+        - jameysharp: the release process creates some text in the auto PR that says to check a CI result, but looks like it has not been passing for some time. Embark need to change their config. Now no Mac CI. In theory Github will eventually have an Arm64 macOS target.
+        - fitzgen: can we hack this for now with an environment variable somewhere? Could add a single test that enables signing the return address. alexcrichton will try and play around with this. 
+        - Moved Amode stuff into ISLE. 
+- avanhatt: 
+    - non-code paper updates
+- jameysharp: 
+    - Doing some work on regalloc2. Want to use parallel moves resolver to improve tail call ABI support. 
+- elliottt:
+    - Experimenting with realloc2 redundant move eliminator. Also looking at better support for minimization of test cases.
```
Comment (cfallin) [inline cranelift/2023/cranelift-08-09.md:56]: s/realloc2/regalloc2/

File: cranelift/2023/cranelift-08-09.md [hunk 2]
```diff
@@ -11,12 +11,58 @@
...
+    - Two problems: linearizing the CFG, and branch/block optimizations. Extended basic blocks are single-in, multiple out. Using these gives some new optimization opportunity, but leads to complexity/changes for standard compiler technology. ~4 years ago Cranelift used this, now moved away from that, Cranelift has an actual CFG including VCode.
+        - Question: can we do branch optimizations in the mid end? E.g. branch constant folding. Nothing precluding us from converting an if to a jump. 
+        - alexcrichton question: the MachBuffer does nice linearization, does it do more than that? Answer: yes, it does 3 optimizations: jump to fall through to just jump, jump over conditional converted to inverted conditional, redirect branch to unconditional branch to remove indirection (plus cleanup if unused). Design deciscion in regalloc2, require all critical edges to be split.
```
Comment (cfallin) [inline cranelift/2023/cranelift-08-09.md:24]: "jump over conditional converted to inverted conditional" -> "conditional jump over unconditional jump converted to inverted conditional"

**Review by cfallin — APPROVED** [review]
Thanks! Two minor fixes, then happy to merge.

File: cranelift/2023/cranelift-08-09.md (hunk 2 above)
Comment (bjorn3) [inline cranelift/2023/cranelift-08-09.md:24]: *decision

File: cranelift/2023/cranelift-08-09.md [hunk 3]
```diff
@@ -11,12 +11,58 @@
...
+        - Are there patterns where early on bytes from libfuzzer correspond more directly to changes in the generated program. It generates only valid code. 
+        - Curious about: anything written up on how to generate inputs such that libfuzzer is better at minimizing? Probably not. A little unfortunate that we derive values from the byte buffer. 
+        - Other option: use structure-aware mutators instead of structure-aware generators. 
+- uweigand:
+    - Looked at tail call stuff, part of their ABI where they pass arguments by indirection. Not sure how that can occur with the tail call ABIs. Question: is there formal specification of the tail call ABI and what data types it supports? fizgen answer: Wasm datatypes, basically, not the fancier Cranelift things that `cg_clif` uses. Right now: arguments prepared somewhere, copied somewhere else. Would have to do pointer updates or generate differently. Right now they hardcode for i128. Could switch to using a vector register for that case. alexcrichton: on RISC-V as well there is indirection for arguments that will need to change.
```
Comment (bjorn3) [inline cranelift/2023/cranelift-08-09.md:64]: The tail_call call conv doesn't need to support struct arguments I think. Wasmtime doesn't use struct arguments and cg_clif can't really use it as that would be ABI incompatible with cg_llvm and cg_gcc which both don't have support for cranelift's version of a tail call calling convention.

**Pull request merged**